### DISCUSSION TITLE
Why isn't the "Text" property of Option<TType> string?

### DISCUSSION THREAD

**Discussion by StevenTCramer** [microsoft/fluentui-blazor#2062] [Q&A]
![image](https://github.com/microsoft/fluentui-blazor/assets/357219/17c6228f-c4cd-457e-be12-8f148b93e0d0)

I was trying to add in a null Option to my list, but I found the Text property of Options to be of TType.  I don't know why this wouldn't always be string.  Especially with the Name Text.

In my case I can't construct a fake object because of validation on its constructor. 

I think this is a bug and if so I can do a PR. Please confirm.

**Comment by vnbaaij** [comment]
Hmm, not sure if it is a bug per se. Think it has always been like that from te beginning. 

In a case like 
```
List<Option<int>> intOptions = new()
    {
        { new Option<int> { Value = 1, Text = 1, Disabled = true } },
        { new Option<int> { Value = 2, Text = 2 } },
        { new Option<int> { Value = 3, Text = 3 } }

    };
}
```
I find it clearer that it is the same type instead of needing to do `{ new Option<int> { Value = 3, Text = "3" } }`

On the other hand it would allow for things like 
```
List<Option<int>> intOptions = new()
{
    { new Option<string> { Value = 1, Text = "One" } },
    { new Option<string> { Value = 2, Text = "Two", Selected = true } },
    { new Option<string> { Value = 3, Text = "Three" } }

};
```

So, not opposed to changing it, but we need to make sure it is not going to be a breaking change.

**Reply by StevenTCramer** [reply]
The use case I want is when I have an object.  In my case I can't create an invalid object as a placeholder.  So I want the Option Value to the Object and the Text to just be what I display. 

I don't know exactly what you define as a breaking change.  The demo's wouldn't compile "as is" without adding `1.ToString() ` 

So if a change breaks the compile it is typically a breaking change.  But some people define differently.

**Reply by vnbaaij** [reply]
We'll list the change in the WhatsNew as a breaking change. Although, I think, very few people are using the `Option` 'option'.

The needee change you describe is expected with the change made.

**Reply by StevenTCramer** [reply]
Thanks again for these components.  You are by far the most responsive opensource maintainer I know.  I can't task switch that well. Very impressive.

**Reply by vnbaaij** [reply]
Maybe I'm just an AI and you'll learn more about me next week at Build?

**Comment by StevenTCramer** [comment]
I went ahead and did a PR #2063   

As I don't see Text being anything other than "Text" making any sense. :)

**Comment by mrpmorris** [comment]
I agree.

- Value is TValue (value of not-currently-known type)
- Text is text, regardless of the value or its type

```
var sexOptions = new List<Option<Sex?>>
{
    { new Option<Sex?> { Value = Sex.Female, Text = "Female, Selected = true  } },
    { new Option<Sex?> { Value = Sex.Male, Text = "Male"} },
    { new Option<Sex?> { Value = null, Text = "I'd rather not tell you" } }
};
```
